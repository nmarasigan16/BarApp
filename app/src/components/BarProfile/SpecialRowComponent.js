import React from 'react';
import {
    View,
    TouchableOpacity
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { colorScheme } from "../../lib/styles/ColorScheme";
import Text from '../general/text/TextComponent';

const SpecialRowComponent = ({special}) => {
    return (
        <View style={{
            flexDirection: 'column',
            margin: 10,
        }}>
            <View style={{
                flexDirection: 'row',
                alignItems: 'center',
            }}>
               <Icon
                   name={'square'}
                   />
                <View style={{
                    flexDirection: 'column',
                    justifyContent: 'center',
                    marginLeft: 20,
                }}>
                    <Text
                        style={{
                            color: colorScheme.accent
                        }}>
                        {special.description}
                    </Text>

                </View>
            </View>
            {special.hasOwnProperty('object') ?
                <View style={{
                    alignSelf: 'center'
                }}>
                    <TouchableOpacity>
                        <Text
                            style={{
                                color: colorScheme.secondaryAccent
                            }}>
                            View {special.object.name}
                        </Text>
                    </TouchableOpacity>
                </View>: null}
        </View>
    )
};

export default SpecialRowComponent;
