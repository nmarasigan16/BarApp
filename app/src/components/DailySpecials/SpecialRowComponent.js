import React from 'react';
import PropTypes from 'prop-types';
import {
    View,
    TouchableOpacity,
    Dimensions,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { colorScheme } from "../../lib/styles/ColorScheme";
import Text from '../general/text/TextComponent';
import Divider from '../general/divider/DividerComponent';

const SpecialRowComponent = ({special}) => {

    const { width, height } = Dimensions.get('window');

    return (
        <View style={{
            marginTop: width/40,
            marginRight: width/30,
            marginLeft: width/30,
        }}>
        <TouchableOpacity>
            <View
                style={{
                    flexDirection: 'column',
                    alignItems: 'center',
                    marginBottom: width/40,
                }}>
                <Text
                    style={{
                        color: colorScheme.accent,
                        fontWeight: 'bold',
                        fontSize: width/15
                    }}>
                    {special.barName}
                </Text>
                <Text
                    style={{
                        color: colorScheme.secondaryAccent,
                        fontSize: width/20
                    }}>
                    {special.description}
                </Text>
            </View>
        </TouchableOpacity>
            <Divider/>
        </View>
    )
};

SpecialRowComponent.propTypes = {
    special: PropTypes.shape({
        barName: PropTypes.string,
        description: PropTypes.string,
        barId: PropTypes.number,
    })
};

export default SpecialRowComponent;