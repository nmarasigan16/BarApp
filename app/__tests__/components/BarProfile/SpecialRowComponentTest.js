import 'react-native';
import React from 'react';
import SpecialRowComponent from '../../../src/components/BarProfile/SpecialRowComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly with an object', () => {
    const special = {
        description: 'Hello there',
        object: {
            name: 'Rail Card'
        }
    };
    const tree = renderer.create(
        <SpecialRowComponent special={special}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders correctly without an object', () => {
    const special = {
        description: 'Hello there',
    };
    const tree = renderer.create(
        <SpecialRowComponent special={special}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
