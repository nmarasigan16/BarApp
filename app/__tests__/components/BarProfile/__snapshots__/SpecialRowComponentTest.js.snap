// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with an object 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Courier New",
            },
            Object {
              "color": "silver",
            },
          ]
        }
      >
        Hello there
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Courier New",
            },
            Object {
              "color": "darkgray",
            },
          ]
        }
      >
        View 
        Rail Card
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly without an object 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Courier New",
            },
            Object {
              "color": "silver",
            },
          ]
        }
      >
        Hello there
      </Text>
    </View>
  </View>
</View>
`;
